text/0000-dot-profile-utility-buildpack.md

```diff
@@ -0,0 +1,156 @@
+
+## Example 2 (file system side effects)
+
+With this `.profile` script:
+
+```
+echo 'hello' >> "$HOME/hello"
+```
+
+The wrapper would not need to set any environment variables, but should maintain the side effect of creating the `$HOME/hello` file.
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+In short, the wrapper should do:
+
+```
+#!/bin/bash
+set -euo pipefail
+source .profile
+env >&3
+```
+
+We acknowledge that the `env` output will need to be properly quoted to be valid TOML.
+We will use or create a tool to handle that formatting.
+
+This will write the environment, including any variables set in `.profile`, to the [`exec.d` output TOML](https://github.com/buildpacks/spec/blob/main/buildpack.md#execd-output-toml).
+Since it also executes the `.profile` script, any side effects will happen.
+This will solve for both of the simple examples above.
+
+On Windows, for `.profile.bat` scripts, we can take the same approach of wrapping the script like:
+
+```
+call .profile.bat
+set >&%CNB_EXEC_D_HANDLE%
+```
+
+Per the [Operating System Conventions in the CNB spec](https://github.com/buildpacks/spec#operating-system-conventions), this buildpack will support scripts compatible with bash version 3 or greater on Linux, and cmd.exe on Windows.
+
+# Migration
+[migration]: #migration
+
+This buildpack is new, so has no inherent migration considerations.
+
+For the related migration concerns for removing shell functionality in general, see the [Migration Path section of RFC 93](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md#migration-path).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This is a new component to maintain and support.
+That being said, this will enable spec and lifecycle simplifications noted in [RFC 93](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md) (where this approach was first suggested).
+Also, [RFC 97](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md) resolves to support and maintain utility buildpacks.
+
+# Alternatives
+[alternatives]: #alternatives
+
+If we do nothing, we introduce a regression in functionality, and force application developers to rework their `.profile` scripts.
+
+
+# Prior Art
+[prior-art]: #prior-art
+
+- Current CNB spec:
+  - `.profile` scripts are part of the buildpack app interface ([spec](https://github.com/buildpacks/spec/blob/main/buildpack.md#app-interface))
+  - `.profile.d/` scripts are part of the buildpack build phase output ([spec](https://github.com/buildpacks/spec/blob/main/buildpack.md#build)).
+
+- Profile scripts:
+  - Heroku supports [`.profile` scripts](https://devcenter.heroku.com/articles/dynos#the-profile-file) and [`.profile.d/` scripts](https://devcenter.heroku.com/articles/buildpack-api#profile-d-scripts).
+  - Cloud Foundry supports [`.profile and .profile.d/ scripts](https://docs.cloudfoundry.org/devguide/deploy-apps/deploy-app.html#profile).
```

Comment:
Does this mean that developers using Cloud Foundry will have a harder migration path if we don't do https://github.com/buildpacks/rfcs/pull/200/files#r797813213 ?

Reply:
Ah, nevermind - the v3 spec doesn't acknowledge profile.d in the app root directory: https://github.com/buildpacks/spec/blob/main/buildpack.md#app-interface

Reply:
Any action needed here in the text?

Reply:
@mboldt I don't think so, b/c we aren't adding `profile.d` scripts here.